### PR TITLE
tests: Add extra positive spec constant

### PR DESCRIPTION
**Pull request by sfricke-samsung**
Wrote test for #3035 but seems to be not properly setting the `shaderInt8` values for `CreateSetSpecConstantDefaultValuePass`... but this might be a `spirv-opt` issue actually

Wanted to post the test so it doesn't get lost

### CODE REVIEW

**Comment by ncesario-lunarg** [comment]
I'm looking into this as well. Looks like this is broken with the existing code (i.e., without the changes in #3035).

**Comment by sfricke-samsung** [comment]
So looked at this with some fresh eyes

See your change in #3035 gives the same `id_value_map` as before the change

The issue is the fact when I look at the SPIR-V before and after `spirv-opt` runs

Before
```swift
          %a = OpSpecConstant %uchar 0
         %19 = OpSpecConstantOp %uint UConvert %a
      %int_1 = OpConstant %int 1
          %b = OpSpecConstant %uint 0
```

After `CreateSetSpecConstantDefaultValuePass`
```swift
          %a = OpConstant %uchar 0 // Test expects to be set to 0x02
         %19 = OpSpecConstantOp %uint UConvert %a
      %int_1 = OpConstant %int 1
          %b = OpConstant %uint 84148994 // 0x05040302
```

So the issue is it not setting the the `shaderInt8` spec constants as expected... but again this is regardless of your change in #3035 which makes me feel it is find to merge that as I look into if this is correct behavior of `spirv-opt`

File: tests/vkpositivelayertests.cpp [hunk 1]
```diff
@@ -13898,3 +13898,199 @@ TEST_F(VkPositiveLayerTest, SpecializationUnused) {
...
+        printf("%s Did not find required device extension %s; skipped.\n", kSkipPrefix, VK_KHR_SHADER_FLOAT16_INT8_EXTENSION_NAME);
+        return;
+    }
+
+    ASSERT_NO_FATAL_FAILURE(InitState(nullptr, &features2));
+    ASSERT_NO_FATAL_FAILURE(InitRenderTarget());
+
+    // glslang currenlty turned the GLSL to
+    //      %19 = OpSpecConstantOp %uint UConvert %a
+    // which causes issue (to be fixed outside scope of this test)
```
Comment (sfricke-samsung) [inline tests/vkpositivelayertests.cpp:13939]: https://github.com/KhronosGroup/glslang/issues/1642

Comment (ncesario-lunarg) [reply tests/vkpositivelayertests.cpp:13939]: @sfricke-samsung For now, targeting Vulkan >=1.2 seems to get things working for this test (i.e., to get https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/master/layers/shader_validation.cpp#L1888 to pick SPIR-V environment 1.5).

Comment (ncesario-lunarg) [reply tests/vkpositivelayertests.cpp:13939]: Ah, I missed you fixed up the assembly to get things working. That works too :)

**Review by ncesario-lunarg — APPROVED** [review]

**Pull request merged**